Microsoft.AVS.VMFS/Microsoft.AVS.VMFS.psm1

```diff
@@ -1241,4 +1241,313 @@ function Get-VmfsHosts {
    Set-Variable -Name NamedOutputs -Value $NamedOutputs -Scope Global    
    Write-Host ""
     
-}
\ No newline at end of file
+}
+
+
+<#
+    .SYNOPSIS
+     This function collects Storage Adapter info for ESXi host(s) in a given vSphere Cluster.
+    
+    .PARAMETER -ClusterName
+     vSphere Cluster Name
+    
+    .EXAMPLE
+     Get-StorageAdapters -ClusterName "vSphere-cluster-001"   
+
+    .INPUTS
+     vSphere Cluster Name
+
+    .OUTPUTS
+     NamedOutputs detailed storage adapters inventory 
+#>
+
+function Get-StorageAdapters {
+    [CmdletBinding()]
+    [AVSAttribute(10, UpdatesSDDC = $false)]
+
+    Param
+    (
+        [Parameter(
+            Mandatory = $true,
+            HelpMessage = 'vSphere Cluster Name')]
+        [String] $ClusterName
+
+    )
+
+    Write-Host "Collecting storage adapters inventory of all ESXi host(s) under vSphere Cluster $($ClusterName)"
+    Write-Host " " ;
+
+    $Cluster = Get-Cluster -Name $ClusterName -ErrorAction Ignore 
+    if (-not $Cluster) {
+        throw "Cluster $($ClusterName) does not exist."
+    }
+      
+    $NamedOutputs = @{}
+    $VmHosts = $Cluster | Get-VMHost 
+    foreach ($VmHost in $VmHosts) {
+        $Adapters = $Null
+        try {
+            $Adapters = Get-VMHostHba -VMHost $VmHost.Name -ErrorAction Ignore     
+        }
+        catch {
+            Write-Error "Failed to collect VMKernel Info on host $($VmHost.Name), continue collecting about rest of the host(s)."
+            continue
+        }
+
+        $StorageAdapters = New-Object System.Collections.ArrayList     
+        if (-not $Adapters) {
+            continue
+        }
+          
+        foreach ($Adapter in $Adapters) {
+            $St = $Adapter | Select-Object -Property * -ExcludeProperty "VMHost" 
+            $StorageAdapters.Add($St) | Out-Null
+
+        } 
+        $NamedOutputs.Add($VmHost.Name.Trim(), ($StorageAdapters | ConvertTo-Json -Depth 10))
+    }
+   
+    if ($NamedOutputs.Count -gt 0) {
+        Write-host $NamedOutputs | ConvertTo-Json -Depth 10
+    }
+    Set-Variable -Name NamedOutputs -Value $NamedOutputs -Scope Global    
+    Write-Host ""
+    
+}
+
+<#
+    .SYNOPSIS
+     This function collects storage vmkernel adapter for ESXi host(s) in a given vSphere Cluster.
+    
+    .PARAMETER -ClusterName
+     vSphere Cluster Name
+    
+    .EXAMPLE
+     Get-VmKernel -ClusterName "vSphere-cluster-001"   
+
+    .INPUTS
+     vSphere Cluster Name
+
+    .OUTPUTS
+     NamedOutputs detailed storage vmkernel adapters inventory. 
+#>
+
+function Get-VmKernels {
```

Comment:
Could this function be renamed to something such as `Get-VmKernelAdapters`?

Reply:
@srajmohan24   I've renamed as you suggested.